packages/lesswrong/components/tagging/TagSubforumPage2.tsx

```diff
@@ -0,0 +1,509 @@
+                  description={`tag ${tag.name}`}
+                  className={classes.description}
+                />
+              </ContentStyles>
+            </div>
+          )}
+        </AnalyticsContext>
+      </div>
+      <div className={classes.centralColumn}>
+        {editing && <TagDiscussionSection key={tag._id} tag={tag} />}
+        {tag.sequence && <TagIntroSequence tag={tag} />}
+        {!tag.wikiOnly && (
+          <AnalyticsContext pageSectionContext="tagsSection">
+            {tag.sequence ? (
+              <SectionTitle title={`Posts tagged ${tag.name}`}>
+                <PostsListSortDropdown value={query.sortedBy || "relevance"} />
+              </SectionTitle>
+            ) : (
+              <div className={classes.tagHeader}>
+                <div className={classes.postsTaggedTitle}>
+                  Posts tagged <em>{tag.name}</em>
+                </div>
+                <PostsListSortDropdown value={query.sortedBy || "relevance"} />
+              </div>
+            )}
+            <PostsList2 terms={terms} enableTotal tagId={tag._id} itemsPerPage={200}>
+              <AddPostsToTag tag={tag} />
+            </PostsList2>
+          </AnalyticsContext>
+        )}
+      </div>
+    </>
+  );
+  
+  const headerComponent = (
+    <div className={classNames(classes.header, classes.centralColumn)}>
+      {query.flagId && (
+        <span>
+          <Link to={`/tags/dashboard?focus=${query.flagId}`}>
+            <TagFlagItem
+              itemType={["allPages", "myPages"].includes(query.flagId) ? tagFlagItemType[query.flagId] : "tagFlagId"}
+              documentId={query.flagId}
+            />
+          </Link>
+          {nextTag && (
+            <span onClick={() => setEditing(true)}>
+              <Link className={classes.nextLink} to={tagGetUrl(nextTag, { flagId: query.flagId, edit: true })}>
+                Next Tag ({nextTag.name})
+              </Link>
+            </span>
+          )}
+        </span>
+      )}
+      <div className={classes.titleRow}>
+        <Typography variant="display3" className={classes.title}>
+          {tag.name}
+        </Typography>
+        {!tag.wikiOnly && !editing && userHasNewTagSubscriptions(currentUser) && (
+          <SubscribeButton
+            tag={tag}
+            className={classes.notifyMeButton}
+            subscribeMessage="Subscribe"
+            unsubscribeMessage="Unsubscribe"
+            subscriptionType={subscriptionTypes.newTagPosts}
+          />
+        )}
+      </div>
+      <Tabs
+        value={tab}
+        onChange={handleChangeTab}
+        className={classes.tabs}
+        textColor="primary"
+        aria-label="select tab"
+        scrollable
+        scrollButtons="off"
+      >
+        <Tab label="Posts" value="posts" />
+        <Tab label="Wiki" value="wiki" />
+      </Tabs>
+    </div>
+  );
+
+  const welcomeBoxComponent = tag.subforumWelcomeText?.html  ? (
```

Comment:
Obviously not blocking or urgent, but with all of these separate components being defined in line it might be worth making a separate `subforum` directory for everything.

Reply:
Yeah I'm planning to do a medium sized refactor when most stuff is done